sqlglot/dialects/exasol.py

```diff
@@ -19,6 +22,15 @@ def _sha2_sql(self: Exasol.Generator, expression: exp.SHA2) -> str:
     return self.func(func_name, expression.this)
 
 
+# https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/trunc%5Bate%5D%20(datetime).htm
+# https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/trunc%5Bate%5D%20(number).htm
+def _parse_trunc(args: t.List) -> exp.DateTrunc | exp.Anonymous:
+    if args and isinstance(args[0], exp.Literal) and args[0].is_number:
+        return exp.Anonymous(this="TRUNC", expressions=args)
+
+    return exp.DateTrunc(unit=seq_get(args, 1), this=seq_get(args, 0))
```

Comment:
This doesn't look right. If we can't *statically* infer the semantics of the `TRUNC[ATE]` function, we should conservatively default to `Anonymous`, not the other way around.